anatomy_of_variables/Anatomy of Variables.md

```diff
@@ -0,0 +1,229 @@
+
+```ruby
+def counter(count)
+    while count <= 10
+        puts count
+        count += 1
+    end
+end
+
+counter(0); #=>0 1 2 3 4 5 6 7 8 9 10
+```
+
+Here is the JavaScript version of the code above.
+
+```JavaScript
+function counter(count){
+    while (count <= 10) {
+       console.log(count);
+        count ++;
+    }
+}
+
+var count = 0;
+counter(count); //=>0 1 2 3 4 5 6 7 8 9 10
+```
+
+**Global Variables**
+
+These are variables that can be used by two or more methods or function. These variables are assigned outside the methods or functions and they can be manipulated within methods or functions.
+
+The example below shows how global variables are assigned and used in Ruby and JavaScript.
+
+```ruby
+@count = 0
+
+def counter
+    while @count <= 10
+        puts @count
+        @count += 1
+    end
+end
+
+counter #=>0 1 2 3 4 5 6 7 8 9 10
+```
+
+
+```JavaScript
+var count = 0;
+
+function counter(count){
+    while (count <= 10) {
+       console.log(count);
+        count ++;
+    }
+}
+
+counter(); //=>0 1 2 3 4 5 6 7 8 9 10
+```
+
+You will notice that `count` was assigned outside the function or method, and it was incremented within the function or method. Also, we need not pass `count` as an argument when calling the method or function.
+
+
+Based on these two types of variables, Ruby has other types of variables. They are:
+
+* Instance Variables
+* Class Variables
+* Class Instant Variables
+* Constants
+
+**Instance Variables**
+
+Instance Variables are global variables that are **ONLY** used in **Instance Methods** within a class. Instance variables in ruby start with a single  `@` sign. Any variable assigned starting with a single `@` sign can be manipulated within an instance method. Just in case, you don't know what an instance method is in ruby, let's take a quick look.
+
+Ruby instance methods are defined with the keyword `def`, then followed by the name of the method and ended with the keyword `end`. Basically, the common methods in ruby are instance methods. These methods can only be called on an instance of the class they belong to.
+The example below shows an instance variable been used in an instance method.
+
+```ruby
+class User
+    def initialize
+        @name = "Adebayo"
+    end
+
+    def greet
+        puts "Hello #{@name}"
+    end
+end
+
+User.new.greet #=> Hello Adebayo
+```
+
+The above example has instance variable `@name` which holds a name `Adebayo`. The instance method `greet` adds `Hello` to the value of `@name`.  This tells us that instance variables act as global variables **within** a class.  
+
+**Class Variables**
+
+Class variables work exactly like instance methods. These varibales are denoted with a double `@` sign, i.e `@@`. These variables are  **ONLY** used in **Class Methods** within a class. Just in case, you don't know what a class method is in ruby, let's take a quick look.
```

Comment:
Class variables can be used in `instance methods`. Infact, they can be used anywhere within the class or it's children. They are the closest to the regular `$variable` global variable.